projects/ngneat/reactive-forms/src/lib/types.ts

```diff
@@ -133,7 +132,9 @@ export type AbstractControlsOf<T extends Obj> = {
  *   ])
  * });
  * */
-export type ControlOf<T> = [T] extends [any[]]
+export type ControlOf<T> = unknown extends T
```

Comment:
I didn't like you used `unknown` to detect `any` type, let's use a safer way for it.
please add and use this utility:
```ts
type ExtractAny<T> = T extends Extract<T, string & number & boolean & object & null & undefined> ? any : never;
```

Reply:
@itayod Excuse me, I'm not sure I got this suggestion 🤔. Could you explain how it would work as expected with a full example? I was unable to make it work properly.

Btw, is there any reason of disliking the use of `unknown`? Do you have any cases where it would break? Please let me know, thanks.

Reply:
@rafaelss95 I actually tried it in my project and it broke my code. About the example there you go:
```ts
type ExtractAny<T> = T extends Extract<T, string & number & boolean & object & null & undefined> ? any : never;

export declare type ControlOf<T> = [T] extends ExtractAny<T> ? AbstractControl<T> : [T] extends [Array<any>] ?
  FormArray<ControlOf<UnwrapArray<T>>>
  : [T] extends [object] ? FormGroup<ControlsOf<T>>
  : FormControl<T>;
```

Reply:
Aha, you wrapped `T`... now I was able to make it work :)

Reply:
I noticed another strange behavior (with both solutions):

![Captura de Tela 2020-11-22 às 11 54 33](https://user-images.githubusercontent.com/11965907/99907100-a0022880-2cb9-11eb-9279-4e49d9b8cc9d.png)
![Captura de Tela 2020-11-22 às 11 54 45](https://user-images.githubusercontent.com/11965907/99907102-a4c6dc80-2cb9-11eb-9145-23ac4641d362.png)

Note that when you have a `ReadonlyArray`, it infers it as a `FormGroup`, while a `FormArray` would be expected.